packages/turbo-repository/rust/src/lib.rs

```diff
@@ -131,4 +135,56 @@ impl Workspace {
 
         Ok(map)
     }
+
+    #[napi]
+    pub async fn affected_packages(&self, files: Vec<String>) -> Result<Vec<Package>, Error> {
+        let mapper = ChangeMapper::new(&self.graph, vec![], vec![]);
+
+        let workspace_root = match AbsoluteSystemPath::new(&self.absolute_path) {
+            Ok(path) => path,
+            Err(e) => return Err(Error::from_reason(e.to_string())),
+        };
+
+        let hash_set_of_paths: HashSet<AnchoredSystemPathBuf> = files
+            .into_iter()
+            .filter_map(|s| {
+                let split = s.split(std::path::MAIN_SEPARATOR).collect::<Vec<&str>>();
+                let xx = workspace_root.join_components(&split);
+                match workspace_root.anchor(&xx) {
+                    Ok(path) => Some(path),
+                    Err(e) => None,
+                }
+            })
+            .collect();
+
+        let package_changes = match mapper.changed_packages(hash_set_of_paths, None) {
+            Ok(changes) => changes,
+            Err(e) => return Err(Error::from_reason(e.to_string())),
+        };
+
+        let packages = match package_changes {
+            PackageChanges::All => self
+                .graph
+                .workspaces()
+                .map(|(name, info)| WorkspacePackage {
+                    name: name.to_owned(),
+                    path: info.package_path().to_owned(),
+                })
+                .collect::<Vec<WorkspacePackage>>(),
+            PackageChanges::Some(packages) => packages.into_iter().collect(),
+        };
+
+        let mut serializable_packages: Vec<Package> = packages
+            .into_iter()
+            .map(|p| {
+                let package_path = workspace_root.resolve(&p.path);
+                Package::new(p.name.to_string(), &workspace_root, &package_path)
+            })
+            .filter(|p| p.name != "//")
```

Comment:
turbo core returns root packages here, but I'm not sure why, and I _think_ we don't care about it. I think global changes are already reflected as `PackageChanges::All`